scalafmt-core/shared/src/main/scala/org/scalafmt/config/ScalafmtConfig.scala

```diff
@@ -156,6 +156,7 @@ case class ScalafmtConfig(
     onTestFailure: String = "",
     encoding: Codec = "UTF-8",
     project: ProjectFiles = ProjectFiles(),
+    fileOverride: Conf.Obj = Conf.Obj.empty,
```

Comment:
I think this can have the type `Map[PathMatcher, ScalafmtConfig]`, we should fail fast if there are errors in the overridden settings instead of crashing multiple times for each individual file

Reply:
how do i make `ScalafmtConfig` the value right when loading the parent config? the value should be a merge of `this` and the custom _subset_ specified on the right, and before `this` has been fully loaded, couldn't do it. also, it should only be applied to the top-level config, not the subset configs.

i do the merge below.

Reply:
It's a good question, let me try to work it out...

Reply:
for now, i will keep it as-is, with the merge happening only when requested.